source/org/zfin/feature/repository/HibernateControlledVocabularyRepository.java

```diff
@@ -18,15 +16,13 @@ public HibernateControlledVocabularyRepository(Class<T> clazz) {
     @Override
     public T getControlledVocabularyTerm(String oboID) {
 
-        Criteria criteria = HibernateUtil.currentSession().createCriteria(clazz);
-        Criteria termCriteria = criteria.createCriteria("term");
-        return (T) termCriteria.add(Restrictions.eq("oboID", oboID)).uniqueResult();
+        Query<T> query = HibernateUtil.currentSession().createQuery("from " + clazz.getSimpleName() + " where term.oboID = :oboID", clazz);
```

Comment:
Is this missing a join on the term table? Something like:

```
    String hql = "from " + clazz.getSimpleName() + " as entity "
                 + "join entity.term as term "
                 + "where term.oboID = :oboID";
```

Reply:
I don't see the difference to my query except mine is more succinct. I don't think we need the alias 'entity' and 'term'.